src/sentry/rules/conditions/event_frequency.py

```diff
@@ -90,7 +98,24 @@ def query_hook(self, event, start, end, environment_id):
     def get_rate(self, event, interval, environment_id):
         _, duration = self.intervals[interval]
         end = timezone.now()
-        return self.query(event, end - duration, end, environment_id=environment_id)
+        result = self.query(event, end - duration, end, environment_id=environment_id)
+        comparison_type = self.get_option("comparisonType", COMPARISON_TYPE_COUNT)
+        if comparison_type == COMPARISON_TYPE_PERCENT:
+            comparison_interval = comparison_intervals[self.get_option("comparisonInterval")][1]
+            comparison_end = end - comparison_interval
+            # TODO: Figure out if there's a way we can do this less frequently. All queries are
+            # automatically cached for 10s. We could consider trying to cache this and the main
+            # query for 20s to reduce the load.
```

Comment:
Out of curiosity, what is the performance impact of this query?

Reply:
It basically means we make two queries instead of one. They're for the same length of time, just time shifted, so in practice when these conditions use a comparison period it will double the load on Clickhouse. So hoping if I can increase the time we cache these queries that it'll work out to have a similar cost.